raiden-ts/src/config.ts

```diff
@@ -40,33 +48,35 @@ export const RaidenConfig = t.readonly(
     }),
     t.partial({
       matrixServer: t.string,
-      logger: t.keyof({
-        ['']: null,
-        trace: null,
-        debug: null,
-        info: null,
-        warn: null,
-        error: null,
-      }),
+      logger: t.union([
+        logLevels,
+        t.partial({
+          prevState: logLevels,
+          action: logLevels,
+          error: logLevels,
+          nextState: logLevels,
+        }),
+      ]),
       pfs: t.union([Address, t.string, t.null]),
       subkey: t.boolean,
     }),
   ]),
 );
 export interface RaidenConfig extends t.TypeOf<typeof RaidenConfig> {}
 
+export const PartialRaidenConfig = t.readonly(
+  t.intersection([t.partial(RaidenConfig.type.types['0'].props), RaidenConfig.type.types['1']]),
```

Comment:
Is that the best way we can have this? I am guessing if I get the code checked on an IDE I can probably see what everything is, but without tooling it is not clear what `types['0'].props` and `types['1']` is supposed to be?

Reply:
Unfortunatelly, I didn't find a way to apply `t.partial` to an already set `t.Mixed`, we need to apply it to the members. The way I did is typesafe and properly typed on an IDE, but by looking at the code, one just needs to know that this is the codec version of `Partial<RaidenConfig>`.
You see above `RaidenConfig` is an intersection of 2 objects: the required part, and the optional/partial part. These are the ones referenced there, by making the required part also optional, and intersecting it with the already partial part.